extras/linter_install_tutorial.md

```diff
@@ -0,0 +1,96 @@
+# Install Linter to Automatically Run Before Git Commit
+
+## About
+
+This section describes how to set up running a liner uploading code onto a GitHub repository (git commit ... -> git push).
+A [linter](https://en.wikipedia.org/wiki/Lint_(software)) is a code analysis tool designed to analyze code and flag for programming errors, bugs and stylistic errors.
+This tool allows one to drastically improve their productivity when writing code for their research projects as stylistic errors and programming errors will automatically be covered through this set-up.
+Disclaimer: The following instructions only work for python code as R doesn't have a working linter. (Subject to change if new information were to come out on this fact.)
+
+## Programs and Packages needed to set up Pipeline
+
+Make sure [conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/download.html) or another python package manager is installed on your computer.
+This pipeline is designed to run on python >=3.5.
+
+This setup will also use the following packages (versions can be changed as needed):
+1. pre-commit==2.9.3
+2. black==20.8b1 
+3. jupytext==1.9.1
```

Comment:
I also have `flake8==3.8.4` here but I didn't explicitly test if its required

Reply:
Don't think it needed to be added, but I'll add it so people have the complete picture. Plus, being able to call flake8 is always a plus if people want to manually check their code.